source/1-SDLC-organization/lib/cdk-bootstrap-template.yml

```diff
@@ -0,0 +1,751 @@
+            Condition:
+              ForAnyValue:StringEquals:
+                "iam:PermissionsBoundary":
+                  - !Sub "arn:aws:iam::${AWS::AccountId}:policy/cdk-${Qualifier}-cfn-exec-permissions-boundary-policy-${AWS::AccountId}-${AWS::Region}"
+                  - !Sub "arn:aws:iam::${AWS::AccountId}:policy/cdk-${Qualifier}-cicd-pipeline-permissions-boundary-policy-${AWS::AccountId}-${AWS::Region}"
+            # IamModifyWithLimitedResourceName
+          - Effect: Allow
+            Action:
+              - iam:GetRolePolicy
+              - iam:DeleteRole
+              - iam:PassRole
+              - iam:TagRole
+              - iam:UntagRole
+              - iam:UpdateRole
+              - iam:UpdateRoleDescription
+              - sts:AssumeRole
+            Resource: "*"
+
+            # AlloBoundaryDeletionForCICDSelfMutate
+          - Effect: Allow
+            Action:
+              - iam:DeleteRolePermissionsBoundary
+            Resource:
+              - !Sub "arn:aws:iam::${AWS::AccountId}:role/*"
+            Condition:
+              "StringEquals":
+                "iam:PermissionsBoundary":
+                  - !Sub "arn:aws:iam::${AWS::AccountId}:policy/cdk-${Qualifier}-cicd-pipeline-permissions-boundary-policy-${AWS::AccountId}-${AWS::Region}"
+
+          ##################################################################
+          # Deny dangerous actions that could escalate privilege or cause security incident
+          ##################################################################
+          # DenyChangeAccountPublicAccess
+          - Effect: Deny
+            Action:
+              - s3:PutAccountPublicAccessBlock
+            Resource: "*"
+
+            # DenyBoundaryPolicyEdit
+          - Effect: Deny
+            Action:
+              - iam:CreatePolicyVersion
+              - iam:DeletePolicy
+              - iam:DeletePolicyVersion
+              - iam:SetDefaultPolicyVersion
+            Resource:
+              - !Sub "arn:aws:iam::${AWS::AccountId}:policy/cdk-${Qualifier}-cfn-exec-permissions-boundary-policy-${AWS::AccountId}-${AWS::Region}"
+              - !Sub "arn:aws:iam::${AWS::AccountId}:policy/cdk-${Qualifier}-cicd-pipeline-permissions-boundary-policy-${AWS::AccountId}-${AWS::Region}"
+
+            # DenyBoundaryDelete
+          - Effect: Deny
+            Action:
+              - iam:DeleteUserPermissionsBoundary
+              - iam:DeleteRolePermissionsBoundary
+            Resource:
+              - !Sub "arn:aws:iam::${AWS::AccountId}:user/*"
+              - !Sub "arn:aws:iam::${AWS::AccountId}:role/*"
+            Condition:
+              "ForAnyValue:StringEquals":
+                "iam:PermissionsBoundary":
+                  - !Sub "arn:aws:iam::${AWS::AccountId}:policy/cdk-${Qualifier}-cfn-exec-permissions-boundary-policy-${AWS::AccountId}-${AWS::Region}"
+                  # Pipeline is self-mutate and it uses this role to delete/update CICD permissions boundary
```

Comment:
Notice that it does not deny removing CICD pipeline PermissionsBoundary . (For the case of self-mutate)

The deny part of CICD PermissionBoundary cannot do this.